rust/lance/src/dataset/transaction.rs

```diff
@@ -396,6 +408,26 @@ impl Transaction {
                     }
                 });
             }
+            Operation::Update {
+                removed_fragment_ids,
+                updated_fragments,
+                new_fragments,
+            } => {
+                final_fragments.extend(maybe_existing_fragments?.iter().filter_map(|f| {
+                    if removed_fragment_ids.contains(&f.id) {
+                        return None;
+                    }
+                    if let Some(updated) = updated_fragments.iter().find(|uf| uf.id == f.id) {
+                        Some(updated.clone())
+                    } else {
+                        Some(f.clone())
+                    }
+                }));
+                final_fragments.extend(Self::fragments_with_ids(
+                    new_fragments.clone(),
+                    &mut fragment_id,
+                ));
```

Comment:
We can do this in a follow-up but we should probably loop through `final_indices` here and remove any that involve any updated columns.

Reply:
I don't think we need any special logic here. Update is a delete + append, and we handle each of those correctly as is. The index will just see the updated rows as deleted ones, until it is re-created.